spyder/plugins/console/plugin.py

```diff
@@ -4,376 +4,129 @@
+    sig_refreshed = Signal()
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
+        return _('Internal console')
 
-        # Find/replace widget
-        self.find_widget = FindReplace(self)
-        self.find_widget.set_editor(self.shell)
-        self.find_widget.hide()
-        if parent is not None:
-            self.register_widget_shortcuts(self.find_widget)
-            self.register_widget_shortcuts(self.shell)
+    def get_icon(self):
+        return self.create_icon('stop')  # FIXME:
 
-        # Main layout
-        btn_layout = QHBoxLayout()
-        btn_layout.setAlignment(Qt.AlignLeft)
-        btn_layout.addStretch()
-        btn_layout.addWidget(self.options_button, Qt.AlignRight)
-        layout = create_plugin_layout(btn_layout)
-        layout.addWidget(self.shell)
-        layout.addWidget(self.find_widget)
-        self.setLayout(layout)
-        
-        # Parameters
-        self.shell.toggle_wrap_mode(self.get_option('wrap'))
-            
-        # Accepting drops
-        self.setAcceptDrops(True)
+    def get_description(self):
+        return _('Internal console running Spyder.')
 
-        # Traceback MessageBox
-        self.error_dlg = None
-        self.error_traceback = ""
-        self.dismiss_error = False
+    def register(self):
+        widget = self.get_widget()
 
-    #------ Private API --------------------------------------------------------
-    def set_historylog(self, historylog):
-        """Bind historylog instance to this console
-        Not used anymore since v2.0"""
-        historylog.add_history(self.shell.history_filename)
-        self.shell.append_to_history.connect(historylog.append_to_history)
+        # Signals
+        widget.sig_edit_goto_requested.connect(self.sig_edit_goto_requested)
+        widget.sig_focus_changed.connect(self.sig_focus_changed)
+        widget.sig_quit_requested.connect(self.sig_quit_requested)
+        widget.sig_refreshed.connect(self.sig_refreshed)
 
-    def set_help(self, help_plugin):
-        """Bind help instance to this console"""
-        self.shell.help = help_plugin
+        # Crash handling
+        previous_crash = self.get_conf_option(
+            'previous_crash',
+            default='',
+            section='main',
+        )
+        if previous_crash:
+            widget.handle_exception(
+                text=previous_crash,
+                is_traceback=True,
+                is_faulthandler_report=True,
+            )
 
-    #------ SpyderPluginWidget API ---------------------------------------------
-    def get_plugin_title(self):
-        """Return widget title"""
-        return _('Internal console')
-    
-    def get_focus_widget(self):
-        """
-        Return the widget to give focus to when
-        this plugin's dockwidget is raised on top-level
-        """
-        return self.shell
+        self.update_font()

```

Comment:
As we discussed, this should be done at the API or main window levels. Do you plan to address that in this PR?

Reply:
Yes, fixed